### ISSUE TITLE
React Native: Adding animated-gif to build.gradle crashes my app

### ISSUE THREAD

**Issue by Mislavoo7** [facebook/fresco#2147]
### Description
Hello,

I followed the instructions: https://facebook.github.io/react-native/docs/image.html#gif-and-webp-support-on-android

and added:

```
dependencies {
    ....
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
    compile 'com.facebook.fresco:animated-gif:1.9.0'  // <--- This
}
```

to my android/app/build.gradle and when I open the emulator the app stops working. No error. 
Emulator is Galaxy Nexus API 23 - Android 6.0 (Google APIs) x86 and the rest of my build.gradle file looks like this:

```
android {
  compileSdkVersion 23
    buildToolsVersion '26.0.2'

    defaultConfig {
      applicationId "com.xxxx"
        minSdkVersion 16
        targetSdkVersion 22
        versionCode 1
        versionName "1.0"
        ndk {
          abiFilters "armeabi-v7a", "x86"
        }
    }
  splits {
    abi {
      reset()
        enable enableSeparateBuildPerCPUArchitecture
        universalApk false  // If true, also generate a universal APK
        include "armeabi-v7a", "x86"
    }
  }
  buildTypes {
    release {
      minifyEnabled enableProguardInReleaseBuilds
        proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
    }
  }
  // applicationVariants are e.g. debug, release
  applicationVariants.all { variant ->
    variant.outputs.each { output ->
      // For each separate APK per architecture, set a unique version code as described here:
      // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
      def versionCodes = ["armeabi-v7a":1, "x86":2]
        def abi = output.getFilter(OutputFile.ABI)
        if (abi != null) {  // null for the universal-debug, universal-release variants
          output.versionCodeOverride =
            versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
        }
    }
  }
}

```

The gif <Image /> has width, height and other styles and looks like this:
```
 render() {
    return(
      <View style={styles.loaderWrapper}>
        <Image
        style={styles.loader}
        source={loadingCircle}
        />
      </View>
    );
  }
```

Am I missing something? 

Sorry to bother you and thank you for your help!

**Comment by oprisnik** [comment]
What React Native version are you using? Only recent versions use Fresco 1.9.0 under the hood and you must use the same version for all Fresco artifacts.

You also mention that it crashes your app. Can you post a stack trace? Without one, it's hard to figure out what's going on.

**Comment by Mislavoo7** [comment]
Using:
react-native-cli: 2.0.1
react-native: 0.55.4

I had to move on and came up with a workaround. Just exported the frames from the gif, render them one by one and it looks like it's animated. :)

I tried to replicate the error but the app isn't crashing anymore, the loader.gif is not rotating and I see no error. Looks like nobody has the same problem except me so I think I made something wrong... The workaround is a good enough solution for me. 

Thank you for your reply if you like close the issue.
